### ISSUE TITLE
SA1013 (ClosingCurlyBracketsMustBeSpacedCorrectly) incorrectly reported after initializer list

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#687]
The following statement incorrectly reports SA1013 on the `}` character:

```csharp
builder.AddRange(new[] { "a", "b" }.ToImmutableArray());
```

**Comment by vweijsters** [comment]
This is an interesting case where a potential new rule might come into play.

I would very much prefer the statement above to look like the code below, in order to make the exact scope of the `ToImmutableArray` method instantly clear. I think it can be beneficial to introduce a new rule to enforce this. How do you guys think about this?

```C#
builder.AddRange((new[] { "a", "b" }).ToImmutableArray());
```

**Comment by Przemyslaw-W** [comment]
I don't have strong opinion. However, if new rule should appear, it should
cover both - collection initializer and object initializer

2015-04-23 20:46 GMT+02:00 Vincent Weijsters <notifications@github.com>:

**Comment by AArnott** [comment]
@vweijsters That would seem to conflict with SA____ that doesn't want parentheses added that are not needed.

**Comment by Noryoko** [comment]
Looks like this was fixed in #1086.
